dev/system-test/pipeline.ts

```diff
@@ -0,0 +1,733 @@
+    });
+  });
+
+  it('testLike', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(like('title', '%Guide%'))
+      .select('title')
+      .execute();
+    expectResults(results, {title: "The Hitchhiker's Guide to the Galaxy"});
+  });
+
+  it('testRegexContains', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(regexContains('title', '(?i)(the|of)'))
+      .execute();
+    expect(results.length).to.equal(5);
+  });
+
+  it('testRegexMatches', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(regexMatch('title', '.*(?i)(the|of).*'))
+      .execute();
+    expect(results.length).to.equal(5);
+  });
+
+  it('testArithmeticOperations', async () => {
+    const results = await randomCol
+      .pipeline()
+      .select(
+        add(Field.of('rating'), 1).as('ratingPlusOne'),
+        subtract(Field.of('published'), 1900).as('yearsSince1900'),
+        Field.of('rating').multiply(10).as('ratingTimesTen'),
+        Field.of('rating').divide(2).as('ratingDividedByTwo')
+      )
+      .limit(1)
+      .execute();
+    expectResults(results, {
+      ratingPlusOne: 5.2,
+      yearsSince1900: 79,
+      ratingTimesTen: 42,
+      ratingDividedByTwo: 2.1,
+    });
+  });
+
+  it('testComparisonOperators', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(
+        and(
+          gt('rating', 4.2),
+          lt(Field.of('rating'), 4.5),
+          neq('genre', 'Science Fiction')
+        )
+      )
+      .select('rating', 'title')
+      .sort(Field.of('title').ascending())
+      .execute();
+    expectResults(
+      results,
+      {rating: 4.3, title: 'Crime and Punishment'},
+      {
+        rating: 4.3,
+        title: 'One Hundred Years of Solitude',
+      },
+      {rating: 4.5, title: 'Pride and Prejudice'}
+    );
+  });
+
+  it('testLogicalOperators', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(
+        or(
+          and(gt('rating', 4.5), eq('genre', 'Science Fiction')),
+          lt('published', 1900)
+        )
+      )
+      .select('title')
+      .sort(Field.of('title').ascending())
+      .execute();
+    expectResults(
+      results,
+      {title: 'Crime and Punishment'},
+      {title: 'Dune'},
+      {title: 'Pride and Prejudice'}
+    );
+  });
+
+  it('testChecks', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(not(Field.of('rating').isNaN()))
+      .select(
+        isNull('rating').as('ratingIsNull'),
+        not(Field.of('rating').isNaN()).as('ratingIsNotNaN')
+      )
+      .limit(1)
+      .execute();
+    expectResults(results, {ratingIsNull: false, ratingIsNotNaN: true});
+  });
+
+  it('testMapGet', async () => {
+    const results = await randomCol
+      .pipeline()
+      .select(
+        Field.of('awards').mapGet('hugo').as('hugoAward'),
```

Comment:
I'd also consider `Field.of('awards.hugo').as('hugoAward')`, or is there some special behavior to `mapGet`?

Does `mapGet()` support dot notation? 
`Field.of('awards').mapGet('hugo.year').as('hugoAwardYear')`
vs
`Field.of('awards').mapGet('hugo').mapGet('year').as('hugoAwardYear')`

Reply:
Good question. It looks like mapGet does not support nested fields (always returns null). 

One benefit is maybe mapGet supports field with special chars better (., /, etc).